### PR TITLE
Bigtable: Admin api re-design

### PR DESCRIPTION
**Pull request by jiren**
- Unit test cases using mock
- Docs with examples

### CODE REVIEW

File: google-cloud-bigtable/google-cloud-bigtable.gemspec [hunk 1]
```diff
@@ -27,7 +28,7 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency "minitest", "~> 5.10"
   gem.add_development_dependency "minitest-focus", "~> 1.1"
   gem.add_development_dependency "minitest-rg", "~> 5.2"
-  gem.add_development_dependency "rubocop", "~> 0.50.0"
+  gem.add_development_dependency "rubocop", "~> 0.56.0"
```
Comment (blowmage) [inline google-cloud-bigtable/google-cloud-bigtable.gemspec]: This dependency version is incompatible with Ruby 2.0, and is why [CI failed on Ruby 2.0](https://circleci.com/gh/GoogleCloudPlatform/google-cloud-ruby/3860#tests/containers/5).

**Comment by quartzmo** [comment]
Build is failing at rubocop:

```sh

lib/google/cloud/bigtable/convert.rb:48:38: C: Use duration.nanos.zero? instead of duration.nanos == 0.

          return duration.seconds if duration.nanos == 0

                                     ^^^^^^^^^^^^^^^^^^^
```

**Comment by blowmage** [comment]
@quartzmo Thoughts on the `TypeError: no implicit conversion of String into Integer` error being raised in the jsondoc task?

**Comment by jiren** [comment]
This error is because of markdown headers parsing using `strscan`  in [kramdown](https://github.com/gettalong/kramdown/blob/b393793050f35758def8c17ab34e92d8dc9f6c62/lib/kramdown/parser/kramdown/header.rb#L48).

Ruby version: 2.0

**Comment by sduskis** [comment]
@jiren: why are there 100 files that changed?

**Comment by quartzmo** [comment]
> Thoughts on the TypeError: no implicit conversion of String into Integer error being raised in the jsondoc task?

@jiren Please rebase your branch on master in order to get this [commit](https://github.com/GoogleCloudPlatform/google-cloud-ruby/commit/9819e686f149016b8b6ebad0ab712c5e7c594bc4) which should fix the broken CI build for Ruby 2.0 for this PR. Thanks!

**Comment by blowmage** [comment]
This branch also has conflicts. Rebasing on master will fix the conflicts and fix the build issue.

**Comment by jiren** [comment]
Closing this pull request and creating new one. https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/2128

**Pull request closed without merging**